geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseIndexAdapter.scala

```diff
@@ -372,43 +372,69 @@ class HBaseIndexAdapter(ds: HBaseDataStore) extends IndexAdapter[HBaseDataStore]
         scan
       }
 
-      val maxRangesPerGroup = {
-        def calcMax(maxPerGroup: Int, threads: Int): Int = {
-          val totalRanges = rangesPerRegionServer.values.map(_.length).sum
-          math.min(maxPerGroup, math.max(1, math.ceil(totalRanges.toDouble / threads).toInt))
-        }
-        if (coprocessor) {
-          calcMax(ds.config.maxRangesPerCoprocessorScan, ds.config.coprocessorThreads)
-        } else {
-          calcMax(ds.config.maxRangesPerExtendedScan, ds.config.queryThreads)
+      rangesPerTable.map { case (table, rangesPerRegionServer) =>
+        val maxRangesPerGroup = {
+          def calcMax(maxPerGroup: Int, threads: Int): Int = {
+            val totalRanges = rangesPerRegionServer.values.map(_.size).sum
+            math.min(maxPerGroup, math.max(1, math.ceil(totalRanges.toDouble / threads).toInt))
+          }
+          if (coprocessor) {
+            calcMax(ds.config.maxRangesPerCoprocessorScan, ds.config.coprocessorThreads)
+          } else {
+            calcMax(ds.config.maxRangesPerExtendedScan, ds.config.queryThreads)
```

Comment:
*blinks*  This opens up quite a bit of tuning...

Reply:
i figured it was a lot easier to tweak some configs and see how they performed vs re-building the code base :)